app/services/open_library/book_search.rb

```
@@ -0,0 +1,19 @@
+module OpenLibrary
+  class BookSearch < Base
+    def self.execute(isbn_number)
+      isbn = "ISBN:#{isbn_number}"
+      response = get('/api/books', query: { bibkeys: isbn }).parsed_response[isbn]
+      if response.nil?
+        "El libro no existe"
```

This is a good way to handle this kind of cases, but it has a couple of problems:
- If the book was't found, the HTTP request status would be `:ok` instead of `:not_found`
- This is a minor problem, you are returning a non translated string in case of any errors. I would recommend to return a hash if you have to

How can we solve this? We could raise an exception, catch it in the controller and return a 404 in case there is no books 🤔